### DISCUSSION TITLE
Enable / disable cursor after alphaTab is loaded

### DISCUSSION THREAD

**Discussion by solveit68** [CoderLine/alphaTab#873] [Q&A]
Hello there,

Unfortunately, I am not able to switch the player cursor in my WPF application on or off at the desired time. I probably have to call a render function, but I don't know which one it is.  

So once loaded, the cursor is either permanently on or off. Since I sometimes only display individual notes for practice purposes, the cursor is very annoying there.
Can someone please give me a tip?

AlphaTab.Settings.Player.EnableCursor = false;/true
AlphaTab.Render ...

Thank you.

**Comment by Danielku15** [comment]
Hi.
Often it is required to call [`UpdateSettings`](https://alphatab.net/docs/reference/settings#:~:text=For.net%20simply%20the%20normal%20classes%20are%20used%20and%20changes%20are%20signaled%20via%20UpdateSettings%20call) to reflect all changes. The settings object is not observed in any way and due to that setting chnages need to be manually flushed down. 

But I double checked and in this case it seems we don't support dynamic change of the cursors to be enabled/disabled 😱 . Also the properties `AlphaTab.BeatCursorFill` and `AlphaTab.BarCursorFill` do not update the colors fully down to the control 😟, 

 **But (!)** There is a workaround, you can just set the color of the cursor to `Brushed.Transparent` through `AlphaTab.BeatCursorFill` and `AlphaTab.BarCursorFill` and it would be quasi disabled visually. Those properties are dependency properties and should update 

I created https://github.com/CoderLine/alphaTab/issues/876 as work item.

**Reply by solveit68** [reply]
Thank you for suggestion the workaround but this also seems not to work.
I've tried this in several combinations:

                AlphaTab.BeatCursorFill = Brushes.Transparent;
                AlphaTab.BarCursorFill = Brushes.Transparent;
            
            //AlphaTab.Settings.Player.EnableCursor = this.DataContext.IsInPlayerMode;
            //AlphaTab.Api.UpdateSettings();
            //AlphaTab.RenderTracks();
            
Any other ideas?
